### PR TITLE
Add Settings smart component, Stripes core utilities

### PR DESCRIPTION
**Pull request by ncovercash**
Adds typings for:
- smart component `Settings`
  - bonus `SettingsProps` interface
- core `StripesContext`
- core `useStripes`
- core `withStripes`
- core `stripesShape`
- core `Stripes` class
  - bonus `StripesType` interface (partially completed)

### CODE REVIEW

File: core/src/Stripes.d.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Requireable } from 'react';
+
+/** Type of the `stripes` object */
+export interface StripesType {
+  /**
+   * Check if the current user has a given permission
+   * @see https://github.com/folio-org/stripes/blob/master/doc/dev-guide.md#testing-for-permission
+   */
+  hasPerm(perm: string): boolean;
```
Comment (zburke) [inline core/src/Stripes.d.ts]: Alas, it is not that simple. `hasPerm` may retun undefined. Its sibling, `hasInterface`, may return undefined OR a boolean OR a string OR 0 (zero).

File: core/src/Stripes.d.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+   * @returns false, if the interface is not present
+   * @returns true, if the interface is available and no specific version was requested
+   * @returns the version string, if the interface and requested version is available
+   * @returns 0, if the interface is available but the requested version is not
+   * @see https://github.com/folio-org/stripes/blob/master/doc/dev-guide.md#testing-for-interfaces
+   */
+  hasInterface(
+    interface: string,
+    version?: string,
+  ): boolean | string | undefined;
```
Comment (zburke) [inline core/src/Stripes.d.ts:28]: Do you need 0/int/whatever TypeScript calls numbers here? You mention it in the comments but I don't see it in the typedef.

**Review by zburke — COMMENTED** [review]
closer

File: core/src/Stripes.d.ts (hunk 2 above)
Comment (ncovercash) [reply core/src/Stripes.d.ts:28]: oh darn missed the number too 🙄

**Review by zburke — APPROVED** [review]
LGTM

**Review by danetsao — APPROVED** [review]

**Pull request merged**